File: chapter-02-auth/javascript/auth-flow.js
/**
 * Chapter 2: Authentication
 * Client Credentials flow
 */

require('dotenv').config();
const { ClientSecretCredential } = require('@azure/identity');
const { Client } = require('@microsoft/microsoft-graph-client');
const { TokenCredentialAuthenticationProvider } = require('@microsoft/microsoft-graph-client/authProviders/azureTokenCredentials');

const tenantId = process.env.TENANT_ID;
const clientId = process.env.CLIENT_ID;
const clientSecret = process.env.CLIENT_SECRET;

async function main() {
    console.log('=== Chapter 2: Authentication ===\n');

    if (!tenantId || !clientId || !clientSecret) {
        console.error('Error: Missing environment variables');
        process.exit(1);
    }

    try {
        const credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
        const authProvider = new TokenCredentialAuthenticationProvider(credential, {
            scopes: ['https://graph.microsoft.com/.default']
        });

        const client = Client.initWithMiddleware({ authProvider });

        const org = await client.api('/organization').get();

        console.log('✓ Connection successful');
        console.log(`  Tenant: ${org.value[0].displayName}`);
        console.log(`  ID: ${org.value[0].id}`);
    } catch (error) {
        console.error('✗ Error:', error.message);
    }
}

main();

File: chapter-02-auth/python/auth_flow.py
"""
Chapter 2: Authentication
Client Credentials example with MSAL
"""

import os
from msgraph.core import GraphClient
from azure.identity import ClientSecretCredential


def main():
    print("=== Chapter 2: Authentication ===\n")

    tenant_id = os.getenv('TENANT_ID')
    client_id = os.getenv('CLIENT_ID')
    client_secret = os.getenv('CLIENT_SECRET')

    if not all([tenant_id, client_id, client_secret]):
        print("Error: Missing environment variables")
        return

    try:
        credential = ClientSecretCredential(
            tenant_id=tenant_id,
            client_id=client_id,
            client_secret=client_secret
        )

        client = GraphClient(credential=credential)
        org = client.get('/organization').json()

        print("✓ Connection successful")
        print(f"  Tenant: {org['value'][0]['displayName']}")
        print(f"  ID: {org['value'][0]['id']}")

    except Exception as e:
        print(f"✗ Error: {e}")


if __name__ == '__main__':
    main()
